opensearch_py_ml/ml_commons/model_access_control.py

```diff
@@ -0,0 +1,87 @@
+from opensearchpy import OpenSearch
+from typing import List
+from opensearch_py_ml.ml_commons.ml_common_utils import ML_BASE_URI
+from opensearch_py_ml.exceptions.ml_commons import InvalidParameterError
+
+ACCESS_MODES = ["public", "private", "restricted"]
+
+
+class ModelAccessControl:
+    API_ENDPOINT = "model_group"
+
+    def __init__(self, os_client: OpenSearch):
+        self.client = os_client
+
+    def register_model_group(
+        self,
+        name: str,
+        description: str,
+        access_mode: str,
+        backend_roles: List[str],
+        add_all_backend_roles=False,
+    ):
+        # if not isinstance(name, str):
```

Comment:
@dhrubo-os , do we need these checks? I was not sure about these. so, commented them out.

Reply:
Let's add those checks. Server (ml-commons) will check that eventually. But we should also have our client side check.